src/creep.prototype.js

```diff
@@ -21,6 +22,9 @@ const creepMatrixCache = {};
 let structureMatrixTick;
 let creepMatrixTick;
 
+let lastTick = Game.time;
+let movingCreeps = {};
```

Comment:
Keeping a hash per tick of all creeps that had `moveTo` or `routeTo` called on them.  We naively assume that all of these creeps will vacate their current position and that their next position is not available.  This actually doesn't work in the case of creeps with fatigue or multiple creeps attempting to move to the same position, but otherwise works well enough